.github/workflows/master-push.yml

```diff
@@ -0,0 +1,227 @@
+name: Prepare Release
+on:
+  push:
+    branches:
+      - "master"
+jobs:
+  prepare:
+      runs-on: ubuntu-latest
+      name: Create Draft Release to store artifacts
+      steps:
+          - name: Checkout
+            uses: actions/checkout@v4
+          - name: Export GITHUB_TOKEN to workspace
+            run: echo "GITHUB_ACCESS_TOKEN=${{ secrets.GITHUB_TOKEN }}" >> $GITHUB_ENV
+          - name: Install ruby
+            uses: ruby/setup-ruby@ec02537da5712d66d4d50a0f33b7eb52773b5ed1
+            with:
+              ruby-version: '3.1.2'
+          - name: Install Gems
+            run: |
+              gem install octokit
+          - name: Create draft release to store artifacts.
+            run: ruby ./scripts/github_release.rb create-draft-release
+  build-docs:
+      runs-on: ubuntu-latest
+      name: Package docs and upload them to the draft release
+      needs: prepare
+      steps:
+          - name: Checkout
+            uses: actions/checkout@v4
+          - name: Get Token
+            id: token
+            uses: yuki0n0/action-appstoreconnect-token@v1.0
+            with:
+                # UUID. Can get from App Store Connect.
+                issuer id: ${{ secrets.APPLE_STORE_CONNECT_ISSUER_ID }}
+                # Key ID. Can get from App Store Connect.
+                key id: ${{ secrets.APPLE_STORE_CONNECT_KEY_ID }}
+                # P8 private key. Can get from App Store Connect.
+                key:  ${{ secrets.APPLE_STORE_CONNECT_API_KEY }}
```

Comment:
Apple recommend to generate tokens instead of creating a long lived one
https://developer.apple.com/documentation/appstoreconnectapi/generating_tokens_for_api_requests#3878467

Reply:
With the changes in #8438 I guess this is irrelevant as we can just pass the secrets directly to the script and skip the separate Get Token step.

Reply:
mmm the script arguments accepts a private key path, we will have to add something to be able to pass the private key as value like we have it stored in GitHub secret, instead of a path in that case

Reply:
but I can add that